src/internal/hooks/use-table-interaction-metrics/__tests__/use-table-interaction-metrics.test.tsx

```diff
@@ -1,52 +1,128 @@
+} satisfies RenderProps;
+
+function renderUseTableInteractionMetricsHook(props: RenderProps) {
+  const elementRef = createRef<HTMLElement>();
 
   const { result, rerender, unmount } = renderHook(useTableInteractionMetrics, {
-    initialProps: { ...defaultProps, ...props },
+    initialProps: { elementRef, ...defaultProps, ...props },
   });
 
   return {
+    tableInteractionAttributes: result.current.tableInteractionAttributes,
     setLastUserAction: (name: string) => result.current.setLastUserAction(name),
-    rerender: (props: RenderProps) => rerender({ ...defaultProps, ...props }),
+    rerender: (props: RenderProps) => rerender({ elementRef, ...defaultProps, ...props }),
     unmount,
   };
 }
 
+function TestComponent(props: RenderProps) {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const { tableInteractionAttributes } = useTableInteractionMetrics({ elementRef, ...defaultProps, ...props });
+  return <div {...tableInteractionAttributes} ref={elementRef} data-testid="element" />;
+}
+
+beforeEach(() => {
+  jest.resetAllMocks();
+  mockPerformanceMetrics();
+});
+jest.useFakeTimers();
+
 describe('useTableInteractionMetrics', () => {
+  let componentMounted: jest.Mock;
+
+  beforeEach(() => {
+    componentMounted = jest.fn();
+
+    setComponentMetrics({
+      componentMounted,
+      componentUpdated: jest.fn(),
+    });
+  });
+
+  test('should emit componentMount event on mount', () => {
+    render(<TestComponent />);
+
+    expect(componentMounted).toHaveBeenCalledTimes(1);
+    expect(componentMounted).toHaveBeenCalledWith({
+      taskInteractionId: expect.any(String),
+      componentName: 'table',
+      componentConfiguration: {},
+    });
+  });
+
+  test('data attribute should be present after the first render', () => {
+    const { getByTestId } = render(<TestComponent />);
+    jest.runAllTimers();
+
+    expect(getByTestId('element')).toHaveAttribute('data-analytics-task-interaction-id');
+  });
+
+  test('data attribute should be present after re-rendering', () => {
+    const { getByTestId, rerender } = render(<TestComponent />);
+    const attributeValueBefore = getByTestId('element').getAttribute('data-analytics-task-interaction-id');
+    rerender(<TestComponent />);
+
+    expect(getByTestId('element')).toHaveAttribute('data-analytics-task-interaction-id');
+
+    const attributeValueAfter = getByTestId('element').getAttribute('data-analytics-task-interaction-id');
+    expect(attributeValueAfter).toBe(attributeValueBefore);
+  });
+
+  test('should not render the attribute during server-side rendering', () => {
+    const markup = renderToStaticMarkup(<TestComponent />);
+
+    expect(markup).toBe('<div data-testid="element"></div>');
+  });
+
   describe('Interactions', () => {
+    let componentUpdated: jest.Mock;
+
+    beforeEach(() => {
+      componentUpdated = jest.fn();
+
+      setComponentMetrics({
+        componentMounted: jest.fn(),
+        componentUpdated,
+      });
+    });
```

Comment:
Why do we even need this? we are already resetting the global stubs in line 51. We can assert against the same stub in each test case?

Reply:
Moved variables to the top.